### PR TITLE
Updated the general style of the style guide to make it more readable

### PR DESCRIPTION
**Pull request by ThibaultKlein**
@htinlinn @paulmiard

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +25,7 @@
 * [Best Practices](BestPractices.md)
 
 
-## Overview ##
+# Overview #
```
Comment (htinlinn) [inline README.md]: Let's get rid of the trailing # from this line as well.

File: README.md [hunk 2]
```diff
@@ -50,20 +51,20 @@ Make sure to consider the resources in the open-source Swift repo; specifically,
...
-## Linter ##
+# Linter
 
 In order to automate many of the rules here, we recommend using [our fork of Swiftlint](https://github.com/prolificinteractive/SwiftLint) in your Swift codebase.
 While the main fork of Swiftlint is based around the GitHub style guide for Swift, our fork has additional rules and corrections for rules specific to our
 style guide that you will not find in the GitHub one.
 
 
-## Standards ##
+# Standards #
```
Comment (htinlinn) [inline README.md]: Here too.

File: README.md [hunk 3]
```diff
@@ -50,20 +51,20 @@ Make sure to consider the resources in the open-source Swift repo; specifically,
...
 
-### Naming Conventions ###
+## Naming Conventions ##
```
Comment (htinlinn) [inline README.md]: And here.

**Review by htinlinn — CHANGES_REQUESTED** [review]
Can we also do a quick search for "Rationale" and make sure that they all have colons after them? Right now, some of them do and some of them don't.

**Review by htinlinn — APPROVED** [review]

**Comment by paulmiard** [comment]
Looks good, let's also update the best practices section?

File: BestPractices.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
-# BEST PRACTICES #
+# BEST PRACTICES
```
Comment (htinlinn) [inline BestPractices.md]: Make this "Best Practices"

File: BestPractices.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 
 This is Prolific's Swift best practices document.
 
 The purpose of these best practices is to help you keep clean and bug free Swift code that is recommended to follow in your daily work.
 
-## Table Of Contents ##
+## Table Of Contents
```
Comment (htinlinn) [inline BestPractices.md]: Table Of Contents uses header level 1, not 2 in the main file. Maybe we change this to `#` and also bump everything else up.

File: BestPractices.md [hunk 6]
```diff
@@ -53,22 +53,22 @@ The standard Swift library provides you different assertion functions that affec
...
 `preconditionFailure` means a fatal error and will terminate in both debug and release mode, except for unchecked builds (`-Ounchecked`), then it will never be executed.
 
-#### fatalError ####
+#### fatalError
 `fatalError` acts like `preconditionFailure` but is not affected by the unchecked build flag. It will always terminate your app in both debug and release mode.
 
-#### Production ####
+#### Production
 
 Although crashing on production is not always the ideal. Often you prefer crashing on debug but in production handle the error with a default value like 0 or empty array. To do so, we recommand implementing a function that takes an optional value as well as a tuple containing the default value of the same type and a error message for the context.
```
Comment (htinlinn) [inline BestPractices.md]: Replace the first two sentences with the following.

```
While forcing a crash in debug builds is acceptable for testing purposes, it is not ideal to have the program exit on assertion failure in production builds. Instead, handling the error with a default value like `0` or empty array might be preferable. 
```

File: BestPractices.md [hunk 7]
```diff
@@ -53,22 +53,22 @@ The standard Swift library provides you different assertion functions that affec
...
 
-#### preconditionFailure ####
+#### preconditionFailure
 `preconditionFailure` means a fatal error and will terminate in both debug and release mode, except for unchecked builds (`-Ounchecked`), then it will never be executed.
 
-#### fatalError ####
+#### fatalError
 `fatalError` acts like `preconditionFailure` but is not affected by the unchecked build flag. It will always terminate your app in both debug and release mode.
 
-#### Production ####
```
Comment (htinlinn) [inline BestPractices.md]: Change this to "Production Builds".

File: BestPractices.md [hunk 8]
```diff
@@ -158,7 +158,8 @@ According to Apple's [documentation](https://developer.apple.com/library/ios/doc
 
 ```Use a weak reference whenever it is valid for that reference to become nil at some point during its lifetime. Conversely, use an unowned reference when you know that the reference will never be nil once it has been set during initialization.```
 
-##### Example #####
+##### Example
```
Comment (htinlinn) [inline BestPractices.md]: Make sure that the subsections here the same number of `#` as subsections in the other sections.

**Review by htinlinn — CHANGES_REQUESTED** [review]

**Review by htinlinn — APPROVED** [review]

**Review by phillfarrugia — APPROVED** [review]

**Comment by paulmiard** [comment]
👍

**Pull request merged**